simdev/module/github/github_api_wrapper.py

```diff
@@ -0,0 +1,186 @@
+            func=self.fetch_stargazers, ignore=["progress", "batch_count", "page_limit"]
+        )
+        self.fetch_starred_repositories = PipelineCache.memory.cache(
+            func=self.fetch_starred_repositories,
+            ignore=["progress", "batch_count", "page_limit"],
+        )
+
+    def fetch_stargazers(
+        self, repo: str, batch_count=100, page_limit=400, progress=None
+    ) -> Set[str]:
+        """
+        Fetch stargazers of the repository
+        :param repo: name (GitHub flavoured. Example: theseems/HseNotebooks)
+        :param batch_count: entries per GitHub API response page
+        :param page_limit: max amount of pages to fetch
+        :param progress: TQDM progress if any
+        :return: set of stargazers
+        """
+        if progress is not None:
+            progress.desc = f"Fetching stargazers for {repo}"
+
+        fixed_url_part = (
+            f"{self._api_url}repos/{repo}/stargazers?per_page={str(batch_count)}&page="
+        )
+        current_page = 0
+        result: Set[str] = set()
+
+        while True:
+            if progress is not None:
+                progress.set_postfix_str("page " + str(current_page + 1))
+            page_response: List[dict] = self._get_request_json(
+                fixed_url_part + str(current_page)
+            )
+            if len(page_response) == 0:
+                break
+            if not isinstance(page_response, list):
+                logging.warning(
+                    "Failure fetching stargazers for %s:"
+                    "\nRequest: %s"
+                    '\nUnknown response: "%s"',
+                    repo,
+                    fixed_url_part + str(current_page),
+                    json.dumps(page_response),
+                )
+                continue
+            for stargazer in page_response:
+                result.add(stargazer["login"])
+            current_page += 1
+            if current_page > page_limit:
+                break
+        return result
+
+    def fetch_starred_repositories(
+        self, user, batch_count=100, page_limit=400, progress=None
+    ) -> Set[str]:
+        """
+        Fetch repositories starred by a user
+        :param user: to fetch starred repositories of
+        :param batch_count: number of entries per GitHub API response's page
+        :param page_limit: max amount of pages to fetch
+        :param progress: TQDM progress
+        :return: set of starred repositories
+        """
+        if progress is not None:
+            progress.desc = f"Fetching starred repositories of {user}"
+
+        fixed_url_part = (
+            f"{self._api_url}users/{user}/starred?per_page={batch_count}&page="
+        )
+        current_page = 0
+        result = set()
+
+        while True:
+            page_response: starred_response_type = None
+            try:
+                if progress is not None:
+                    progress.set_postfix_str("page " + str(current_page + 1))
```

Comment:
it's duplicated logic - you can move it to separate function like `update_progress_bar` - and your functions will be simplified.
Please check other places with duplicated logic